Source/WebCore/page/FrameView.cpp

```diff
@@ -3693,7 +3693,9 @@ void FrameView::performPostLayoutTasks()
     // FIXME: We should not run any JavaScript code in this function.
     LOG(Layout, "FrameView %p performPostLayoutTasks", this);
     updateHasReachedSignificantRenderedTextThreshold();
-    frame().selection().updateAppearanceAfterLayout();
+
+    if (auto& selection = frame().selection(); selection.isFocusedAndActive())
+        selection.updateAppearanceAfterLayout();
```

Comment:
Can we file a followup bug to remove this entirely?

Reply:
Filed https://bugs.webkit.org/show_bug.cgi?id=247041, and added a FIXME here referencing that bug.